app/src/test/java/com/x8bit/bitwarden/data/autofill/processor/AutofillCipherProviderTest.kt

```diff
@@ -155,6 +160,51 @@ class AutofillCipherProviderTest {
             assertFalse(result.await())
         }
 
+    @Suppress("MaxLineLength")
+    @Test
+    fun `isVaultLocked when there is an active user should wait for pending unlocking to finish and return the locked state for that user when it times out`() =
+        runTest {
+            every { authRepository.activeUserId } returns ACTIVE_USER_ID
+            mutableVaultStateFlow.value = listOf(
+                VaultUnlockData(
+                    userId = ACTIVE_USER_ID,
+                    status = VaultUnlockData.Status.UNLOCKING,
+                ),
+            )
+
+            val result = async {
+                autofillCipherProvider.isVaultLocked()
+            }
+
+            testScheduler.runCurrent()
+            assertFalse(result.isCompleted)
+
+            testScheduler.advanceUntilIdle()
+            assertTrue(result.isCompleted)
+
+            assertTrue(result.await())
+        }
+
+    @Suppress("MaxLineLength")
+    @Test
+    fun `getCardAutofillCiphers when unlocked should return empty list when retrieving ciphers times out`() =
+        runTest {
+            mutableCiphersStateFlow.value = DataState.Loading
+
+            // Test
+            val actual = async {
+                autofillCipherProvider.getCardAutofillCiphers()
+            }
+
+            assertFalse(actual.isCompleted)
```

Comment:
This statement is not meaningful if you haven't called `runCurrent` first. So we should always do that before checking for completeness.